### PR TITLE
ENH: added routines to convert ddp between full and SA formulations

### PR DESCRIPTION
**Pull request by sglyon**
Sometimes I find it helpful to have convenient, even if not optimally efficient, routines for converting between the SA and full formulations. 

This PR implements such routines.

### CODE REVIEW

File: quantecon/markov/ddp.py [hunk 1]
```diff
@@ -456,6 +456,73 @@ def _check_action_feasibility(self):
...
+        Notes
+        -----
+        If this instance is already in full form then it is returned
+        un-modified
+
+        """
+        if self._sa_pair:
+            ns = self.num_states
+            na = self.a_indices.max() + 1
+            R = np.zeros((ns, na))
```
Comment (oyamad) [inline quantecon/markov/ddp.py]: Maybe this should be `np.full((ns, na), -np.inf)`?

File: quantecon/markov/ddp.py [hunk 2]
```diff
@@ -456,6 +456,73 @@ def _check_action_feasibility(self):
...
+        Notes
+        -----
+        If this instance is already in SA form then it is returned
+        un-modified
+        """
+
+        if self._sa_pair:
+            return self
+        else:
+            s_ind, a_ind = np.where(np.isfinite(self.R))
```
Comment (oyamad) [inline quantecon/markov/ddp.py]: Better to write `self.R > -np.inf` instead of `np.isfinite(self.R)`?
(`self.R` may contain `np.inf`.)

**Comment by oyamad** [comment]
A comment on terminology:
I guess the terminologies should be consistent between the  code and the documentation: "product formulation" should be replaced "full formulation" in the documentation, or vice versa.

(My view is that an SA formulation is already a "full" description of the model when the sets `A(s)` of available actions at state `s` are given excluding infeasible actions.)

**Comment by sglyon** [comment]
Thanks for the review. I 100% agree on all accounts and will make the changes in a few hours when I'm back at a computer

**Comment by oyamad** [comment]
In the test code, maybe better to assign `-np.inf` to some entry in `R` and `0` in `Q`.
Otherwise the code looks good to me.

**Comment by sglyon** [comment]
Good call @oyamad -- I've added these tests.

**Comment by mmcky** [comment]
Thanks @sglyon and @oyamad for review. This looks good to me. I will leave open one more day and then merge.

**Comment by mmcky** [comment]
Thanks @sglyon.

**Pull request merged**